spec/consensus/light-client.md

```diff
@@ -316,3 +316,70 @@ func Backwards(h1,h2) bool {
   return (hash(h2) == old.Header.hash)
  }
 ```
+
+### Failure Detection Module
+
+The light client must be able to detect failures in the trust assumptions.
+Ideally, it should never trust a header, which was forged by malicious
+validators.
+
+### Connectivity requirement
+
+To meet the above requirement, it must be connected to at least one honest
+full node. This is sufficient so long as there's no fork on the main chain.
+
+If there is a fork on the main chain, it means that two full nodes have decided
+on two different headers for the same height. Then the requirement would at
+least require that the lite client is **connected to one honest full node on
+each branch of the fork**.
+
+_Assumption_: no more than `2f` of the nodes are malicious (`f+1` are honest).
+otherwise, they could create more forks.
+
+In practice, this means connecting to 1 or more geographically distributed full
+nodes (called witnesses), which belong to different companies. Note this number
+does not include the primary full node (called primary), which is used for
+obtaining new headers.
+
+_Remark_: we can't guarantee all witnesses won't follow the same branch of a
+fork. to guarantee that, we'll need the light client to be connected to +1/3 of
+nodes, which is impractical.
+
+### Evidence detection
+
+After the light client verifies a new header (`H`) it received from primary, it
+should cross-check `H` with the headers from all witnesses. Cross-checking
+means comparing hashes of the headers. If any two hashes (or more) diverge,
+there's a fork (on the main chain OR phantom fork targeting this light client).
+
+The light client will then need to validate the header it got from a witness
+(`H1`) and verify it's signed by at least one trusted validator. If either of
+these fails, it should disconnect from the offending witness.
+
+1. Equivocation
+
+  If some validator double signed, the light client should form & submit a
+  `DuplicateVoteEvidence`.
+
+  After doing so, the light client must stop its operation. The operator will be
```

Comment:
Should the light client continue to work say if H1 is not signed by +1/3 (no fork on the main chain)?

Reply:
I am not sure to understand the question: does this mean H1 obviously is a attack on the lite client only? I guess if the lite client detects that it may continue and look for the correct header of that height. ... but again, I am not sure I understood the question.

Reply:
> does this mean H1 obviously is a attack on the lite client only?

Exactly

Reply:
Not sure we should bother with the light client detecting eg individual double signs if they're not part of a successful attack.

So I think there's really only two error modes here:
1) Faulty full node: mark them as bad and stop talking to them, but otherwise continue
2) Successful +1/3 attack: submit evidence and halt, wait for human intervention

Sounds like you're describing a case where they receive a conflicting header that does not fully verify but does contain a double sign from a validator. Since the commit doesn't fully verify, this is a faulty full node. Technically there is a faulty validator in here but I guess they would just go unpunished ... we could concern ourselves with trying to capture this down the road if we really want but I don't think it's a big issue if we don't.

Reply:
> Sounds like you're describing a case where they receive a conflicting header that does not fully verify but does contain a double sign from a validator

yep

> Technically there is a faulty validator in here but I guess they would just go unpunished 

Okay, I think we can skip this for now.